static/js/user_pill.js

```diff
@@ -146,7 +148,9 @@ exports.set_up_typeahead_on_pills = function (input, pills, update_func, source)
         updater(user) {
             exports.append_user(user, pills);
             input.trigger("focus");
-            update_func();
+            if (opts.update_func && typeof opts.update_func === "function") {
```

Comment:
Instead of doing this function-check in `if` statement I'd have done something like this,
```js
            if (opts.update_func) {
                if (opts.update_func !== "function") {
                    blueslip.error(<error message here>);
                }
                opts.update_func();
            }
```
But I'm not sure we are doing these sort of checks at other places. But it's definitely a good thing to do. Maybe, typescript will take care of this in future.

Reply:
A small commit message for this commit 9cec9ecfd41dc1d2204b36406c4eb5190a9be66f is appreciated. Can be 1-3 lines explaining that it's a preliminary commit.

Reply:
> But I'm not sure we are doing these sort of checks at other places

I see, I checked at other typeahead places and this check wasn't necessary as javascript already raises a `TypeError`.